src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+            }
+            catch (TimeoutException)
+            {
+                LogInfo.Log($"{nameof(GetRxStatus)}: {nameof(TimeoutException)} in {nameof(SpiReadRegister)}", LogLevel.Debug);
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool GetIrqStatus(Span<byte> irqStatus)
+        {
+            LogInfo.Log($"{nameof(GetIrqStatus)}", LogLevel.Debug);
+            if (irqStatus.Length != 4)
+            {
+                throw new ArgumentException($"{nameof(irqStatus)} needs to be 4 bytes");
+            }
+
+            try
+            {
+                SpiReadRegister(Register.IRQ_STATUS, irqStatus);
+                LogInfo.Log($"{nameof(GetIrqStatus)}: {nameof(irqStatus)}: {BitConverter.ToString(irqStatus.ToArray())}", LogLevel.Debug);
+            }
+            catch (TimeoutException)
+            {
+                LogInfo.Log($"{nameof(GetIrqStatus)}: {nameof(TimeoutException)} in {nameof(SpiReadRegister)}", LogLevel.Debug);
+                return false;
+            }
+
+            return true;
+        }
+
+        #endregion
+
+        #region SPI primitives
+
+        private void SpiWriteRegister(Command command, Register register, Span<byte> data)
+        {
+            if (data.Length != 4)
+            {
+                throw new ArgumentException($"Register data must be 4 bytes");
+            }
+
+            Span<byte> toSend = stackalloc byte[2 + data.Length];
+            toSend[0] = (byte)command;
+            toSend[1] = (byte)register;
+            data.CopyTo(toSend.Slice(2));
+            SpiWrite(toSend);
+        }
+
+        private void SpiReadRegister(Register register, Span<byte> readBuffer)
+        {
+            Span<byte> toSend = stackalloc byte[2]
+            {
+                (byte)Command.READ_REGISTER,
+                (byte)register
+            };
+            SpiWrite(toSend);
+            SpiRead(readBuffer);
+        }
+
+        private void SpiWriteRead(Span<byte> toSend, Span<byte> toRead)
+        {
+            SpiWrite(toSend);
+            SpiRead(toRead);
+        }
+
+        private void SpiWrite(Span<byte> toSend)
+        {
+            // Both master and slave devices must operate with the same timing.The master device
+            // always places data on the MOSI line a half cycle before the clock edge SCK, in order for
+            // the slave device to latch the data.
+            // The BUSY line is used to indicate that the system is BUSY and cannot receive any data
+            // from a host.Recommendation for the BUSY line handling by the host:
+            // 1.Assert NSS to Low
+            // 2.Perform Data Exchange
+            // 3.Wait until BUSY is high
+            // 4.Deassert NSS
+            // 5.Wait until BUSY is low
+            // Wait for the PN8150 to be ready
+            int timeout = TimeoutWaiting;
```

Comment:
I'd prefer we measured time instead of cycles. This will vary too greatly depending on where it runs